packages/app/src/cli/services/dev.ts

```diff
@@ -162,7 +162,7 @@ async function dev(options: DevOptions) {
 
   const additionalProcesses: OutputProcess[] = []
 
-  if (localApp.extensions.theme.length > 0) {
+  if (localApp.extensions.theme.length > 0 && !isSpinEnvironment()) {
```

Comment:
I'd recommend telling Shopifolks about this limitation. Otherwise, they'll think it's a bug, and we are adding a support burden on us. 

```
Your application has theme app extensions that won't be served. There's a limitation that causes them not to work in Spin environments and the team is working on fixing it - Link.
```